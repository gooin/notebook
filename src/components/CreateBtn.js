import React from 'react';
import {Button, Row, Col} from "antd";

const CreateBtn = ({onCreate}) => {
    return (
        <div style={{margin: "1rem 0"}}>
            <Row>
                <Col span={4}/>
                <Col span={16}>
                    <Button
                        className={"creat-item-button"}
                        onClick={onCreate}
                        type="primary"
                        block={true}
                    >创建一条新的记录</Button>
                </Col>
                <Col span={4}/>
            </Row>
        </div>
    );
};

export default CreateBtn;
