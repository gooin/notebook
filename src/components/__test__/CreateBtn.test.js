import React from 'react';
import {shallow} from 'enzyme';
import CreateBtn from "../CreateBtn";
import toJson from 'enzyme-to-json';


const props = {
    onCreate: jest.fn()
}


describe('test CreateBtn component', () => {
    let wrapper = shallow(<CreateBtn {...props}/>)
    it('should Button rendered', () => {
        expect(wrapper.find('Button').length).toBe(1)
    })

    it('check snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('should button clicked', () => {
        wrapper.find('Button').simulate('click')
        expect(props.onCreate).toHaveBeenCalled()
    })

})