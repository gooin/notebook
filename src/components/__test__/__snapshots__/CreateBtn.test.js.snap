// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CreateBtn component check snapshot 1`] = `
<div
  style={
    Object {
      "margin": "1rem 0",
    }
  }
>
  <Row>
    <Col
      span={4}
    />
    <Col
      span={16}
    >
      <Button
        block={true}
        ghost={false}
        htmlType="button"
        loading={false}
        onClick={[MockFunction]}
        type="primary"
      >
        创建一条新的记录
      </Button>
    </Col>
    <Col
      span={4}
    />
  </Row>
</div>
`;
